File: spring-backend/src/main/java/com/steiner/vblog/model/article/ArticleShortcut.java
package com.steiner.vblog.model.article;

import com.steiner.vblog.model.Category;
import com.steiner.vblog.model.Tag;
import com.steiner.vblog.model.User;
import jakarta.annotation.Nonnull;
import jakarta.annotation.Nullable;

import java.sql.Timestamp;
import java.util.List;

@lombok.Builder
public class ArticleShortcut {
    @Nonnull
    public static ArticleShortcut fromArticle(@Nonnull Article article) {
        return ArticleShortcut.builder()
                .id(article.id)
                .title(article.title)
                .summary(article.summary)
                .category(article.category)
                .author(article.author)
                .publishDate(article.publishDate)
                .editTime(article.editTime)
                .status(article.status)
                .tags(article.tags)
                .build();
    }

    public int id;

    @Nonnull
    @lombok.NonNull
    public String title;

    @Nonnull
    @lombok.NonNull
    public String summary;

    @Nullable
    public Category category;

    @Nonnull
    @lombok.NonNull
    public User author;

    @Nonnull
    @lombok.NonNull
    public java.sql.Timestamp publishDate;

    @Nonnull
    @lombok.NonNull
    public java.sql.Timestamp editTime;

    @Nonnull
    @lombok.NonNull
    public ArticleStatus status;

    @Nonnull
    @lombok.NonNull
    public List<Tag> tags;
}

File: backend/src/main/kotlin/com/steiner/vblog/model/ArticleShortcut.kt
package com.steiner.vblog.model

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
class ArticleShortcut(
    val id: Int,
    val title: String,
    val summary: String,
    val category: Category?,
    val author: User,
    val publishDate: LocalDateTime,
    val editTime: LocalDateTime,
    val status: ArticleStatus,
    val tags: List<Tag>
) {
    companion object {
        @JvmStatic
        fun fromArticle(article: Article): ArticleShortcut {
            return ArticleShortcut(
                id = article.id,
                title = article.title,
                summary = article.summary,
                category = article.category,
                author = article.author,
                publishDate = article.publishDate,
                editTime = article.editTime,
                status = article.status,
                tags = article.tags
            )
        }
    }

}